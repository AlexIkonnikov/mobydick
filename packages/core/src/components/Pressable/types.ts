import {PressableProps as DefaultPressableProps} from 'react-native';

export type PressableProps = DefaultPressableProps;
