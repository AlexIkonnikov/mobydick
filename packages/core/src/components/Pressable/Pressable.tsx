import React, {forwardRef} from 'react';
import {Pressable as DefaultPressable} from 'react-native';

import {IView} from '../View';

import {IPressableProps} from './types';

const Pressable = forwardRef<IView, IPressableProps>((props, ref) => {
  return <DefaultPressable ref={ref} {...props} />;
});

export default Pressable;
