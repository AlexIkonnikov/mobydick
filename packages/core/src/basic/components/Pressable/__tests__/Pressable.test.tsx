import {render} from '@testing-library/react-native';

import Pressable from '../Pressable';

describe('@lad-tech/mobydick-core/Pressable', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<Pressable />);
    expect(toJSON()).toMatchSnapshot();
  });
});
