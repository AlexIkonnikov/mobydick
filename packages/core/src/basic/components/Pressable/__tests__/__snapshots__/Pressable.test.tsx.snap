// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-core/Pressable renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
/>
`;
